go/codepipeline-build-deploy/codepipeline-build-deploy.go

```diff
@@ -0,0 +1,326 @@
+	triggerCodeBuild := lambda.NewFunction(stack, jsii.String("BuildLambda"), &lambda.FunctionProps{
+		Architecture: lambda.Architecture_ARM_64(),
+		Code:         lambda.AssetCode_FromAsset(jsii.String("./lambda/"), nil),
+		Handler:      jsii.String("trigger-build.handler"),
+		Runtime:      lambda.Runtime_NODEJS_18_X(),
+		Environment: &map[string]*string{
+			"REGION":                 jsii.String(os.Getenv("CDK_DEFAULT_REGION")),
+			"CODEBUILD_PROJECT_NAME": jsii.String(*initialBuild.ProjectName()),
+		},
+		InitialPolicy: &[]iam.PolicyStatement{
+			iam.NewPolicyStatement(&iam.PolicyStatementProps{
+				Actions: &[]*string{
+					jsii.String("codebuild:StartBuild"),
+				},
+				Effect: iam.Effect_ALLOW,
+				Resources: &[]*string{
+					jsii.String(*initialBuild.ProjectArn()),
+				},
+			}),
+		},
+	})
+
+	// Triggers a Lambda function using AWS SDK
+	triggerLambda := custom.NewAwsCustomResource(stack, jsii.String("BuildLambdaTrigger"), &custom.AwsCustomResourceProps{
+		InstallLatestAwsSdk: jsii.Bool(true),
+		Policy: custom.AwsCustomResourcePolicy_FromStatements(&[]iam.PolicyStatement{
+			iam.NewPolicyStatement(&iam.PolicyStatementProps{
+				Actions: &[]*string{
+					jsii.String("lambda:InvokeFunction"),
+				},
+				Effect: iam.Effect_ALLOW,
+				Resources: &[]*string{
+					triggerCodeBuild.FunctionArn(),
+				},
+			}),
+		}),
+		OnCreate: &custom.AwsSdkCall{
+			Service:            jsii.String("Lambda"),
+			Action:             jsii.String("invoke"),
+			PhysicalResourceId: custom.PhysicalResourceId_Of(jsii.String("id")),
+			Parameters: map[string]*string{
+				"FunctionName":   triggerCodeBuild.FunctionName(),
+				"InvocationType": jsii.String("Event"),
+			},
+		},
+		OnUpdate: &custom.AwsSdkCall{
+			Service: jsii.String("Lambda"),
+			Action:  jsii.String("invoke"),
+			Parameters: map[string]*string{
+				"FunctionName":   triggerCodeBuild.FunctionName(),
+				"InvocationType": jsii.String("Event"),
+			},
+		},
+	})
+
+	// Creates VPC for the ECS Cluster
+	clusterVpc := ec2.NewVpc(stack, jsii.String("ClusterVpc"), &ec2.VpcProps{
+		IpAddresses: ec2.IpAddresses_Cidr(jsii.String("10.45.0.0/16")),
+	})
+
+	// Deploys the cluster VPC after the initial image build triggers
+	clusterVpc.Node().AddDependency(triggerLambda)
+
+	// Creates a new blue Target Group
+	targetGroupBlue := elb.NewApplicationTargetGroup(stack, jsii.String("BlueTargetGroup"),
+		&elb.ApplicationTargetGroupProps{
+			TargetGroupName: jsii.String("alb-blue-tg"),
+			TargetType:      elb.TargetType_IP,
+			Port:            jsii.Number(80),
+			Vpc:             clusterVpc,
+		},
+	)
+
+	// Creates a new green Target Group
+	targetGroupGreen := elb.NewApplicationTargetGroup(stack, jsii.String("GreenTargetGroup"),
+		&elb.ApplicationTargetGroupProps{
+			TargetGroupName: jsii.String("alb-green-tg"),
+			TargetType:      elb.TargetType_IP,
+			Port:            jsii.Number(80),
+			Vpc:             clusterVpc,
+		},
+	)
```

Comment:
Add some documentation about what target groups are, with links to relevant information.